Index: compiz.fix_1120009/src/main.cpp
===================================================================
--- compiz.fix_1120009.orig/src/main.cpp	2013-02-14 12:41:55.389902417 +0800
+++ compiz.fix_1120009/src/main.cpp	2013-02-14 12:43:31.150377272 +0800
@@ -38,6 +38,11 @@
 #include "privatescreen.h"
 #include "privatestackdebugger.h"
 
+namespace
+{
+bool autoAddCCP = true;
+}
+
 void
 CompManager::usage ()
 {
@@ -52,6 +57,7 @@
             "  --debug               Enable debug mode\n"
             "  --version             Show the program version\n"
             "  --help                Show this summary\n"
+	    "  --no-auto-add-ccp     Do not automatically load the ccp plugin\n"
             , programName);
 }
 
@@ -116,6 +122,10 @@
 	    if (i + 1 < argc)
 		clientId = argv[++i];
 	}
+	else if (!strcmp (argv[i], "--no-auto-add-ccp"))
+	{
+	    autoAddCCP = false;
+	}
 	else if (*argv[i] == '-')
 	{
 	    compLogMessage ("core", CompLogLevelWarn,
@@ -151,23 +161,25 @@
 
     modHandler = new ModifierHandler ();
 
-    if (!initialPlugins.empty ())
-    {
-	CompOption::Value::Vector list;
-        CompOption::Value         value;
+    CompOption::Value::Vector list;
+    CompOption::Value         value;
 	CompOption                *o = screen->getOption ("active_plugins");
 
+    if (!initialPlugins.empty ())
+    {
 	foreach (CompString &str, initialPlugins)
 	{
 	    value.set (str);
 	    list.push_back (value);
 	}
+    }
+    else if (autoAddCCP)
+	list.push_back ("ccp");
 
 	value.set (CompOption::TypeString, list);
 
 	if (o)
 	    o->set (value);
-    }
 
     if (!screen->init (displayName))
 	return false;
