Index: compiz/src/screen.cpp
===================================================================
--- compiz.orig/src/screen.cpp	2015-09-18 04:41:44.071761222 +0200
+++ compiz/src/screen.cpp	2015-09-18 04:47:21.284758934 +0200
@@ -3294,10 +3294,19 @@
 	     * This is so that we can detect taps on individual modifier
 	     * keys, and know to cancel the tap if <modifier>+k is pressed.
 	     */
-	    if (!(currentState & CompAction::StateIgnoreTap))
+	    int minCode, maxCode;
+	    XDisplayKeycodes (screen->dpy(), &minCode, &maxCode);
+
+	    if ((currentState & CompAction::StateIgnoreTap))
+	    {
+		KeyCode code_p = XKeysymToKeycode(screen->dpy(), XK_p);
+
+		for (k = minCode; k <= maxCode; k++)
+		    if (k != code_p)
+			grabUngrabOneKey (modifiers | ignore, k, grab);
+	    }
+	    else
 	    {
-		int minCode, maxCode;
-		XDisplayKeycodes (screen->dpy(), &minCode, &maxCode);
 		for (k = minCode; k <= maxCode; k++)
 		    grabUngrabOneKey (modifiers | modifierForKeycode | ignore, k, grab);
 	    }
